src/QuorumTools/Cluster.hs

```diff
@@ -316,7 +349,11 @@ createNode genesisJsonPath gid = do
     let acctKey = forceAcctKey $ cEnv ^? clusterAccountKeys . ix gid
     installAccountKey gid acctKey
     eid <- requestEnodeId gid
-    mkGeth gid eid
+    mRaftPort <- raftPort gid
```

Comment:
The control flow is kind of complicated. Mind explaining it to me (in person is fine)? Here we ask the geth node for its enode id so we can parse out its port. Why don't we have the port already? I'm sure this makes sense but I want to understand all the steps.

Reply:
Yeah it's a little weird -- I'll add a comment.  In this scenario, when we ask the geth node for `admin.nodeInfo.enode`, it comes back with a `discport`, but no `raftport`. We are trying to construct an `EnodeId` here that includes a `raftport` querystring parameter. So we get the enode-id-sans-raftport, the port from the `raftPort` function, and build the new EnodeId with `addRaftPort`.

Reply:
Actually let's see if we can get `admin.nodeInfo.enode` to return a raftport on the geth side.

Reply:
I talked to Joel about this. This is difficult, but will be easier/cleaner to implement once we make `--raft` imply `--nodiscover` on the geth side. Punting for now